client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -116,6 +118,12 @@ public void readerOffline(String readerId, Position lastPosition) {
         return synchronizer.getState().getStreamNames();
     }
 
+    @Override
+    public ReaderGroupConfig getReaderGroupConfig() {
+        synchronizer.fetchUpdates();
```

Comment:
In this case, `synchronizer.fetchUpdates()` is called in order to get the correct `ReaderGroupConfig`. Without it, calling for the config right after creating a ReaderGroup, lets say during `resetReaderGroup`, was throwing a null pointer exception on the `getConfig` call.

Reply:
What I see is, all getXXX() calls on ReaderGroupImpl do fetchUpdates() before returning the value of that object, with possibly the intent of getting the most recent value from the StateSynchronizer Stream. The intent here is the same.
@tkaitchuck  What concerns do you see with doing fetchUpdates here?

Reply:
i think it is correct and consistent to call fetchupdates as RG can be "reset". after an update to rg config, if this method is invoked, it will be consistent in its response.